File: Platform-Transport_Backend/PlatformTransport/src/main/java/com/platformtrasnport/platformtransport/model/OffreTransport.java
    package com.platformtrasnport.platformtransport.model;

    import com.platformtrasnport.platformtransport.model.enul.OffreStatus;
    import com.platformtrasnport.platformtransport.model.enul.TypeOffreTransport;
    import jakarta.persistence.*;
    import lombok.Getter;
    import lombok.Setter;

    import java.io.Serializable;
    import java.time.LocalDate;

    @Entity
    @Getter
    @Setter
    public class OffreTransport implements Serializable {

        @Id
        @GeneratedValue(strategy = GenerationType.IDENTITY)
        private Long id;

        @Enumerated(EnumType.STRING)
        private TypeOffreTransport typeOffreTransport;

        private String pointDepart;
        private String destination;

        private LocalDate dateOffre;

        private int nombrePlaces;
        private float prix;
        private String imgUrl;
        private OffreStatus status;

        @ManyToOne
        @JoinColumn(name = "employeur_id")
        private Employeur employeur;


    }

File: Platform-Transport_FrontEnd/PlatformTransport/src/app/core/models/OffreTransport.ts
import { TypeOffreTransport } from './enum/TypeOffreTransport';
import { OffreStatus } from './enum/OffreStatus';
import { RejectionReason } from './enum/RejectionReason';

export interface OffreTransport {
  id?: number;
  typeOffreTransport: TypeOffreTransport;
  pointDepart: string;
  destination: string;
  dateOffre: string;
  nombrePlaces: number;
  prix: number;
  imgUrl: string;
  status?: OffreStatus;
  rejectionReason?: RejectionReason;
}
